specs/menu/menu.md

```diff
@@ -0,0 +1,210 @@
+
+### Risks and Challenges
+
+#### Nesting:
+
+Infinite nesting of menus (submenus) presents a challenge for keyboard navigation and styling.
+
+```html
+<fast-menu>
+    <fast-menu-item>Menu item one</fast-menu-item>
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        Menu item three
+        <fast-menu slot="item">
+            <fast-menu-item>Nested item one</fast-menu-item>
+            <fast-menu-item>Nested item two</fast-menu-item>
+            <fast-menu-item>
+                Nested item three
+                <fast-menu slot="item">
+                    <fast-menu-item>Nested item one</fast-menu-item>
+                    <fast-menu-item>Nested item two</fast-menu-item>
+                </fast-menu>
+            </fast-menu-item>
+        </fast-menu>
+    </fast-menu-item>
+    <fast-menu-item>Menu item four</fast-menu-item>
+</fast-context-menu>
+```
+
+#### Icon support:
+
+Consider a scenario where we have the following menu - some menu items have glyphs, others do not:
+```html
+<fast-menu>
+    <fast-menu-item>Menu item one</fast-menu-item>
+    <fast-menu-item>Menu item two</fast-menu-item>
+    <fast-menu-item>
+        <fancy-icon slot="icon"></fany-icon>
+        Menu item three
+    </fast-menu-item>
+</fast-context-menu>
+```
+The third item in the menu has a glyph, which would inset the text from the other menu items. To ensure that the content of this menu item aligns with the rest of the items, all items need to visually shift to ensure alignmnet of the menu content. This presents an interesting challenge for how we will communicate these changes to ensure the styling is appropriately applied. Ultimately, the parent just needs to know that *one* of the items below has a menu item with the glyph.
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/context-menu)
+- [Material UI](https://material-ui.com/components/menus/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/menus/)
+- [Ant Design - menu](https://ant.design/components/menu/)
+- [Ant Design - dropdown](https://ant.design/components/menu/)
+- [Atlassian - menu](https://atlaskit.atlassian.com/packages/core/menu)
+- [Atlassian - dropdown menu](https://atlaskit.atlassian.com/packages/core/dropdown-menu)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/tree-view)
+
+---
+
+## Design
+
+### API
+
+*The key elements of the component's public API surface:*
+
+**Menu**
+*Component name:*
+- `fast-menu`
+
+*Attributes:*
+- extends HTML Element attributes
+
+*Slots:*
+- default slot for items
+
+*Events*
+- none
+
+**Menu Item**
+*Component name:*
+- `fast-menu-item`
+
+*Attributes:*
+- role - an enum representing the menu items' role
```

Comment:
Do we need to build in radio and checkbox? or is there simply a slot that this or other components could be put into? Does it affect the menu item itself in any particular way?

Reply:
Do we need a menu item separator?

Reply:
My thought is that it's up to users to manage their markup in some sense. The menu takes items with the roles of `menuitem`, `menuitemcheckbox`, `menuitemradio` and I think it would also take the a simple `<hr />` or div with a role of separator to manage the separation - at least that is the current working model. Menu items with a checkbox role can stand up next to standard menu items, but menu items with a radio role should be contained in a group. Menu items with checkbox and radio functionality both use aria-checked and visually they would look the same (with a checkmark). The primary difference is that in a group of items with `menuitemradio` only one can be selected.

Reply:
Maybe we have a special slotted content style for hr?

Reply:
yeah, this seems reasonable.